src/Microsoft.OpenApi/Services/OpenApiDocumentComparer.cs

```diff
@@ -14,21 +15,35 @@ public class OpenApiDocumentComparer : OpenApiComparerBase<OpenApiDocument>
         /// Executes comparision against source and target <see cref="OpenApiDocument"/>.
         /// </summary>
         /// <param name="sourceDocument">The source.</param>
-        /// <param name="targetDocument">The target</param>
+        /// <param name="targetDocument">The target.</param>
         /// <param name="comparisonContext">Context under which to compare the source and target.</param>
         public override void Compare(
             OpenApiDocument sourceDocument,
             OpenApiDocument targetDocument,
             ComparisonContext comparisonContext)
         {
-            comparisonContext.GetComparer<OpenApiPaths>().Compare(
-                sourceDocument.Paths,
-                targetDocument.Paths,
-                comparisonContext);
+            WalkAndCompare(
+                comparisonContext,
+                OpenApiConstants.Paths,
+                () => comparisonContext
+                    .GetComparer<OpenApiPaths>()
+                    .Compare(sourceDocument.Paths, targetDocument.Paths, comparisonContext));
+
+            WalkAndCompare(
+                comparisonContext,
+                OpenApiConstants.Components,
+                () => comparisonContext
+                    .GetComparer<OpenApiComponents>()
+                    .Compare(sourceDocument.Components, targetDocument.Components, comparisonContext));
+
+            WalkAndCompare(
+                comparisonContext,
+                OpenApiConstants.Components,
```

Comment:
>Components [](start = 33, length = 10)

Servers

Reply:
good catch :-)

---
In reply to: [220656072](https://github.com/Microsoft/OpenAPI.NET/pull/316#discussion_r220656072) [](ancestors = 220656072)